### PR TITLE
updated benchmarking for validators-manager pallet

### PR DESCRIPTION
**Pull request by ivan-cholakov**
Updated validators-manager `benchmarks.rs` so that tests compile

### CODE REVIEW

File: pallets/validators-manager/src/benchmarking.rs [hunk 1]
```diff
@@ -100,14 +101,24 @@ fn setup_action_data<T: Config>(
...
         <T as Config>::AccountToBytesConvert::into_bytes(&action_account_id),
     ));
 
     #[cfg(test)]
     T::CandidateTransactionSubmitter::reserve_transaction_id(&candidate_tx.clone()).unwrap();
-    #[cfg(not(test))]
-    T::CandidateTransactionSubmitter::set_transaction_id(&candidate_tx.clone(), eth_transaction_id);
+    // #[cfg(not(test))]
+    // T::CandidateTransactionSubmitter::set_transaction_id(&candidate_tx.clone(),
+    // eth_transaction_id);
```
Comment (thadouk) [inline pallets/validators-manager/src/benchmarking.rs:121]: Is this an incomplete change?

File: pallets/validators-manager/src/tests_voting_deregistration.rs [hunk 2]
```diff
@@ -130,20 +131,13 @@ fn approve_validator_action(
...
         ValidatorManager::sign_validators_action_for_ethereum(&context.action_id).unwrap();
     return ValidatorManager::approve_validator_action(
         RawOrigin::None.into(),
         context.action_id,
         validator.clone(),
         approval_signature,
-        context.record_deregister_validator_calculation_signature.clone(),
-    )
-}
-
```
Comment (thadouk) [inline pallets/validators-manager/src/tests_voting_deregistration.rs:146]: these changes look off

File: pallets/ethereum-transactions/src/lib.rs [hunk 3]
```diff
@@ -627,7 +627,6 @@ pub trait CandidateTransactionSubmitter<AccountId> {
     ) -> DispatchResult;
 
     /// Sets a transaction Id. This is only enabled for benchmarks
-    #[cfg(feature = "runtime-benchmarks")]
```
Comment (thadouk) [inline pallets/ethereum-transactions/src/lib.rs:630]: Why was this removed?

File: pallets/ethereum-transactions/src/lib.rs [hunk 4]
```diff
@@ -695,7 +694,6 @@ impl<T: Config> CandidateTransactionSubmitter<T::AccountId> for Pallet<T> {
         Ok(())
     }
 
-    #[cfg(feature = "runtime-benchmarks")]
```
Comment (thadouk) [inline pallets/ethereum-transactions/src/lib.rs:698]: Why was this removed?

File: pallets/ethereum-transactions/src/lib.rs (hunk 3 above)
Comment (ivan-cholakov) [reply pallets/ethereum-transactions/src/lib.rs:630]: because otherwise the compiler outputs an error `set_transaction_id not found on trait...`

File: pallets/ethereum-transactions/src/lib.rs (hunk 4 above)
Comment (ivan-cholakov) [reply pallets/ethereum-transactions/src/lib.rs:698]: same as the above

File: pallets/validators-manager/Cargo.toml [hunk 5]
```diff
@@ -68,4 +68,5 @@ runtime-benchmarks = [
     "frame-benchmarking/runtime-benchmarks",
     "frame-support/runtime-benchmarks",
     "frame-system/runtime-benchmarks",
+	"pallet-ethereum-transactions/runtime-benchmarks"
```
Comment (thadouk) [inline pallets/validators-manager/Cargo.toml:71]:
```suggestion
    "pallet-ethereum-transactions/runtime-benchmarks"
```

**Review by thadouk — APPROVED** [review]
